src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -153,10 +153,10 @@ private void consume() throws Exception {
         _sensors._consumeError.record();
         continue;
       }
-      int partition = record.partition();
-      long index = (Long) avroRecord.get(DefaultTopicSchema.INDEX_FIELD.name());
+      int partition = (int) avroRecord.get(DefaultTopicSchema.PARTITION_FIELD.name());
```

Comment:
After thinking more about the problem yesterday, I think there may be problem with identifying partition using message payload. But before I explain my concern, it is probably better if you can think through this as well since you are the owner of this feature. Can you try to answer the following questions:

- How can we use MirroredPipelineMonitor to detect problem in the source cluster?
- How can we use MirroredPipelineMonitor to detect problem in the second cluster?
- Is there any impact if we identify partition using message payload instead using the RecordMetadata? What it change guarantee of any existing metrics?
- What is the problem we are solving? Is there better solution to address the problem?